src/xmagics/xassist.cpp

```diff
@@ -279,6 +279,53 @@ namespace xcpp
         }
     };
 
+    std::string escape_special_cases(const std::string& input)
+    {
+        std::string escaped;
+        for (char c : input)
+        {
+            switch (c)
+            {
+                case '\\':
+                    escaped += "\\\\";
+                    break;
+                case '\"':
+                    escaped += "\\\"";
+                    break;
+                case '\n':
+                    escaped += "\\n";
+                    break;
+                case '\t':
+                    escaped += "\\t";
+                    break;
+                case '\r':
+                    escaped += "\\r";
+                    break;
+                case '\b':
+                    escaped += "\\b";
+                    break;
+                case '\f':
+                    escaped += "\\f";
+                    break;
+                default:
+                    if (c < 0x20 || c > 0x7E)
+                    {
+                        // Escape non-printable ASCII characters and non-ASCII characters
+                        std::array<char, 7> buffer{};
+                        std::stringstream ss;
+                        ss << "\\u" << std::hex << std::setw(4) << std::setfill('0') << (c & 0xFFFF);
+                        escaped += ss.str();
+                    }
+                    else
+                    {
+                        escaped += c;
+                    }
+                    break;
+            }
+        }
+        return escaped;
+    }
+
```

Comment:
> anything involving disrupting the JSON to call the LLM.

Hmm, so should there be a test required here as to check here to check this claim ?

Reply:
> Hmm, so should there be a test required in this PR to check this claim ?

I did try but got in a predicament where `escape_special_cases("""{}") == "\"\"{}"` yields false but works in the kernel. So I tested them manually and pushed the code